### PR TITLE
Error-reporting command

### PR DESCRIPTION
**Pull request by rosen-vladimirov**
Add error-reporting command that should be used in a similar manner as feature-usage-tracking command. The difference is that during installation, the exceptions tracking will be enabled without prompt (keep current behavior that we are tracking exceptions). It can be disabled later by using `$ appbuilder error-reporting disable`.
Refactor analytics-service in order to "hide" some methods from the interface and to simplify adding of new analytics commands and variables.

Use correct parameters when starting simulator.
Add unit tests for analytics-service and simulator-service.

**Rename feature-usage-tracking to usage-reporting**

http://teampulse.telerik.com/view#item/291773
and http://teampulse.telerik.com/view#item/296481

### CODE REVIEW

File: test/analytics-service.ts [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+import helpersLib = require("../lib/common/helpers");
+import hostInfoLib = require("../lib/common/host-info");
+let assert = require("chai").assert;
+
+let lastTrackedFeatureNameAndValue = "";
+let lastSavedSettingNameAndValue = "";
+let lastTrackedExceptionMsg = "";
+let lastUsedEqatecSettings: any;
+let isEqatecStopCalled = false;
+let shouldSetUserThrowException = false;
```
Comment (rosen-vladimirov) [inline test/analytics-service.ts]: remove this variable, it is not used

**Comment by rosen-vladimirov** [comment]
ping @ikoevska  for the help changes (all of them are in the second commit)

File: docs/man_pages/general/autocomplete-disable.md [hunk 2]
```diff
@@ -16,7 +16,8 @@ Command | Description
 [autocomplete](autocomplete.html) | Asks for confirmation if command line autocompletion should be enabled for bash and zsh.
 [autocomplete-status](autocomplete-status.html) | Prints the current status of your command-line completion settings.
 [autocomplete-enable](autocomplete-enable.html) | Enables command-line completion for bash and zsh shells.
-[feature-usage-tracking](feature-usage-tracking.html) | Configures anonymous usage statistics tracking for the NativeScript command-line interface.
+[feature-usage-tracking](feature-usage-tracking.html) | Configures anonymous usage statistics tracking for the AppBuilder CLI.
```
Comment (ikoevska) [inline docs/man_pages/general/autocomplete-disable.md]: Telerik AppBuilder CLI

File: docs/man_pages/general/autocomplete-disable.md [hunk 3]
```diff
@@ -16,7 +16,8 @@ Command | Description
...
+[exceptions-tracking](exceptions-tracking.html) | Configures anonymous exceptions tracking for the Telerik AppBuilder CLI.
 [help](help.html) | Lists the available commands or shows information about the selected command.
-[doctor](doctor.html) | Checks your system and reports potential problems which may prevent developing with AppBuilder CLI.
+[doctor](doctor.html) | Checks your system and reports potential problems which may prevent developing with Telerik AppBuilder CLI.
```
Comment (ikoevska) [inline docs/man_pages/general/autocomplete-disable.md]: Checks your system for configuration problems which might prevent the Telerik AppBuilder CLI from working properly.

**Comment by ikoevska** [comment]
Add the new commands to index.md

**Comment by rosen-vladimirov** [comment]
As we've discussed - the files "feature-usage-tracking-disable", "feature-usage-tracking-enable", "feature-usage-tracking-status" are never loaded from command line when executing `appbuilder feature-usage-tracking enable -h`. As the command is not hierarchical, we show the default page (feature-usage-tracking.md file). The same is valid for exceptions-tracking command :)

**Comment by ligaz** [comment]
Shouldn't we use the word `errors` instead of `exceptions`? I'm not convinced either which one is better :confused:

**Comment by rosen-vladimirov** [comment]
I've used the word "exceptions" as it is used in our other clients. Here's the screenshot from Graphite:
graphite_exceptions_tracking

As this command is used for the same functionality, I've decided to use the word exceptions and make it as close as possible with our `feature-usage-tracking` command.

**Comment by teobugslayer** [comment]
run ci

**Comment by ikoevska** [comment]
I'm transferring my comment from Skype here. After some research, it seems that the industry standard for "exception tracking" is actually "error report". It might be a good idea to stick with this term since it is less intimidating than "tracking" and implies pro-active engagement on the side of the user.

**Comment by teobugslayer** [comment]
error report looks good to me

**Comment by rosen-vladimirov** [comment]
Ok, I'll change it :)

**Comment by rosen-vladimirov** [comment]
Ping @teobugslayer , @ikoevska  - I've used `error-report`, please review the changes :)

**Comment by ikoevska** [comment]
Ok, I feel a bit stupid here but I'd rather we go with error-reporting for the name of the command :D I have somehow failed to communicate that error report is the general term not the command name we should be using. Btw, since we're going this way, why not rename feature-usage-tracking to usage-reporting?

**Comment by rosen-vladimirov** [comment]
ok, error-report is now error-reporting and feature-usage-tracking is now usage-reporting

File: test/analytics-service.ts [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+		};
+		lastTrackedFeatureNameAndValue = "";
+		lastTrackedExceptionMsg = "";
+		lastSavedSettingNameAndValue = "";
+		isEqatecStopCalled = false;
+		lastUsedEqatecSettings = {};
+	});
+
+	after(() => {
+		global._eqatec = originalEqatec;
```
Comment (teobugslayer) [inline test/analytics-service.ts]: Good, good!

**Comment by teobugslayer** [comment]
:+1: but I have remarks on the code in common lib.

**Comment by ikoevska** [comment]
From what I'm seeing docs/man_pages/general/feature-usage-tracking.md is not renamed. But the command help file is referenced as usage-reporting.md (which does not seem to exist). Is this correct o.O

**Comment by rosen-vladimirov** [comment]
Nice catch @ikoevska , I've fixed it. Thank you :smile_cat:

File: docs/man_pages/general/usage-reporting.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
 
-Configures anonymous usage statistics tracking for the Telerik AppBuilder CLI.
+Configures anonymous usage reporting for the Telerik AppBuilder CLI.
 
 All data gathered is used strictly for improving the product and will never be used to identify or contact you.
 
 ### Attributes
-`<Command>` extends the `feature-usage-tracking` command. You can set the following values for this attribute.
+`<Command>` extends the `usage-reporting` command. You can set the following values for this attribute.
 * `status` - Shows the current configuration for anonymous usage tracking for the Telerik AppBuilder CLI.
```
Comment (ikoevska) [inline docs/man_pages/general/usage-reporting.md]: the dreaded tracking is here. Please, change it to reporting for status, disable and enable.

**Comment by rosen-vladimirov** [comment]
ping @ikoevska  - I've fixed the help, could you please take a look at it one last time

**Comment by ikoevska** [comment]
:+1:

File: docs/man_pages/index.md [hunk 6]
```diff
@@ -11,7 +11,8 @@ Command | Description
 [login](general/login.html) | Logs you in the Telerik Platform.
 [logout](general/logout.html) | Logs you out from the Telerik Platform.
 [user](general/user.html) | Prints information about the currently logged in user.
-[feature-usage-tracking](general/feature-usage-tracking.html) | Configures anonymous feature usage tracking.
+[usage-reporting](general/usage-reporting.html) | Configures anonymous usage reporting.
+[error-reporting](general/error-reporting.html) | Configures anonymous error reporting.
 [autocomplete](general/autocomplete.html) | Manages your command-line completion settings.
```
Comment (rosen-vladimirov) [inline docs/man_pages/index.md]: add doctor command here

Comment (rosen-vladimirov) [inline docs/man_pages/index.md]: done

**Comment by Fatme** [comment]
:+1:

**Pull request merged**